pkg/channel/http/http_channel.go

```diff
@@ -76,13 +81,31 @@ func (h *Channel) InvokeMethod(invokeRequest *channel.InvokeRequest) (*channel.I
 	}
 	req.Header.SetMethod(method)
 
+	if invokeRequest.Metadata != nil {
+		if corID, ok := invokeRequest.Metadata[tracing.CorrelationID]; ok {
+			req.Header.Set(tracing.CorrelationID, corID)
+		}
+	}
+	var span tracing.TracerSpan
```

Comment:
Can lines 89 - 97 be encapsulated in the diagnostics package? only like 98 has something to do with the channel.